app/lib/frontend/backend.dart

```diff
@@ -591,53 +597,103 @@ class GCloudPackageRepository extends PackageRepository {
 
   @override
   Future addUploader(String packageName, String uploaderEmail) async {
-    return _withAuthenticatedUser((String userEmail) {
-      return db.withTransaction((Transaction T) async {
-        final packageKey = db.emptyKey.append(models.Package, id: packageName);
-        final models.Package package = (await T.lookup([packageKey])).first;
+    await _withAuthenticatedUser((String userEmail) async {
+      final packageKey = db.emptyKey.append(models.Package, id: packageName);
+      final models.Package package = (await db.lookup([packageKey])).first;
 
-        // Fail if package doesn't exist.
-        if (package == null) {
-          await T.rollback();
-          throw new GenericProcessingException(
-              'Package "$package" does not exist');
-        }
+      _validateNewUploader(userEmail, package, uploaderEmail);
```

Comment:
This call isn't idempotent..., why not?

If the person being added as uploader is already added, why don't we just return 200 ok?

If we're trying to make restful APIs we should aim for idempotency.

----
I know this changes semantics from what they used to be. But is there a reason we didn't have idempotent APIs?

Reply:
Ah. You are right, we could do a silent success without sending an email.